pallets/payment/src/tests.rs

```diff
@@ -1389,6 +1389,185 @@ fn voucher_call_deprecated_upload_payer_err() {
+            None,
+            true,
+            100,
+        ));
+        let voucher_id = get_last_voucher_id();
+        let voucher_account_id = voucher_id.cast::<AccountIdOf<Test>>();
+
+        assert_eq!(
+            Balances::free_balance(voucher_account_id),
+            voucher_initial_balance
+        );
+
+        // Creating a RuntimeCall that should be free for caller.
+        let call = RuntimeCall::GearVoucher(VoucherCall::call {
+            voucher_id,
+            call: PrepaidCall::DeclineVoucher,
+        });
+
+        // Creating simulation of weight params for call to calculate fee.
+        let len = 100usize;
+        let fee_length = LengthToFeeFor::<Test>::weight_to_fee(&Weight::from_parts(len as u64, 0));
+
+        let weight = Weight::from_parts(1_000, 0);
+        let fee_weight = WeightToFeeFor::<Test>::weight_to_fee(&weight);
+
+        let call_fee = fee_length + fee_weight;
+
+        // Pre-dispatch of the call.
+        let pre = CustomChargeTransactionPayment::<Test>::from(0)
+            .pre_dispatch(&BOB, &call, &info_from_weight(weight), len)
+            .unwrap();
+
+        // Bob hasn't paid fees.
+        assert_eq!(Balances::free_balance(BOB), bob_initial_balance);
+
+        // But the voucher has.
+        assert_eq!(
+            Balances::free_balance(voucher_account_id),
+            voucher_initial_balance - call_fee,
+        );
+
+        // Validator hasn't received fee yet.
+        assert_eq!(
+            Balances::free_balance(BLOCK_AUTHOR),
+            validator_initial_balance
+        );
+
+        // Post-dispatch of the call.
+        assert_ok!(CustomChargeTransactionPayment::<Test>::post_dispatch(
+            Some(pre),
+            &info_from_weight(weight),
+            &default_post_info(),
+            len,
+            &Ok(())
+        ));
+
+        // Validating balances and validator reward.
+        assert_eq!(Balances::free_balance(BOB), bob_initial_balance);
+        assert_eq!(
+            Balances::free_balance(voucher_account_id),
+            voucher_initial_balance - call_fee,
+        );
+        assert_eq!(
+            Balances::free_balance(BLOCK_AUTHOR),
+            validator_initial_balance + call_fee,
+        );
+    });
+}
+
+#[test]
+fn voucher_call_decline_payer_expired_err() {
+    new_test_ext().execute_with(|| {
+        // Snapshot of the initial data.
+        let bob_initial_balance = Balances::free_balance(BOB);
+        let voucher_initial_balance = 1_000_000_000;
+
+        // Issuing a voucher to Bob from Alice.
+        assert_ok!(GearVoucher::issue(
+            RuntimeOrigin::signed(ALICE),
+            BOB,
+            voucher_initial_balance,
+            None,
+            false,
+            100,
+        ));
+        let voucher_id = get_last_voucher_id();
+        let voucher_account_id = voucher_id.cast::<AccountIdOf<Test>>();
+
+        System::set_block_number(System::block_number() + 1_000);
```

Comment:
Isn't it better to call `run_to_block`?

Reply:
There we dont care about any hooks or any other activities, except changing bn. Similar hack used in many other places

Reply:
Lets say, test time optimisation